src/serializer/ResidualHeapSerializer.js

```diff
@@ -695,6 +697,107 @@ export class ResidualHeapSerializer {
+          let descriptorValue = descriptor.value;
+
+          if (descriptorValue !== undefined) {
+            return descriptorValue;
+          }
+        }
+      }
+    }
+    return null;
+  }
+
+  _serializeValueReactElementChild(child: Value): BabelNode {
+    const expr = this.serializeValue(child);
+
+    // TODO: what if the child is an array but serializes to a reference?
+    if (t.isArrayExpression(expr)) {
+      applyKeysToNestedArray((expr: any), true);
+      return expr;
+    }
+    if (expr === null) {
+      return t.jSXExpressionContainer(t.jSXEmptyExpression());
+    }
+    if (t.isStringLiteral(expr) || t.isNumericLiteral(expr)) {
+      return t.jSXText((expr: any).value + "");
+    }
+    if (t.isJSXElement(expr)) {
+      return (expr: any).value;
+    }
+    return t.jSXExpressionContainer(expr);
+  }
+
+  _serializeValueReactElement(val: ObjectValue): BabelNodeExpression {
+    let objectProperties: Map<string, any> = val.properties;
+    let typeValue = this._getPropertyValue(objectProperties, "type");
+    let keyValue = this._getPropertyValue(objectProperties, "key");
+    let refValue = this._getPropertyValue(objectProperties, "ref");
+    let propsValue = this._getPropertyValue(objectProperties, "props");
+
+    invariant(typeValue !== null, "JSXElement type of null");
+
+    let identifier = convertExpressionToJSXIdentifier(this.serializeValue(typeValue));
+    let attributes = [];
+    let children = [];
+
+    if (keyValue !== null) {
+      let keyExpr = this.serializeValue(keyValue);
+      if (keyExpr.type !== "NullLiteral") {
+        attributes.push(convertKeyValueToJSXAttribute("key", keyExpr));
+      }
+    }
+
+    if (refValue !== null) {
+      let refExpr = this.serializeValue(refValue);
+      if (refExpr.type !== "NullLiteral") {
+        attributes.push(convertKeyValueToJSXAttribute("ref", refExpr));
+      }
+    }
+
+    if (propsValue instanceof ObjectValue) {
+      this.serializedValues.add(propsValue);
+      // have to case propsValue to ObjectValue or Flow complains that propsValues can be null/undefined
+      for (let [key, propertyBinding] of (propsValue: ObjectValue).properties) {
+        let desc = propertyBinding.descriptor;
+        if (desc === undefined) continue; // deleted
+
+        invariant(key !== "key" && key !== "ref", `"${key}" is a reserved prop name`);
+
+        if (key === "children" && desc.value !== undefined) {
+          let childrenValue = desc.value;
+          if (childrenValue instanceof ArrayValue) {
+            this.serializedValues.add(childrenValue);
+            let childrenLength = this._getPropertyValue(childrenValue.properties, "length");
+            let childrenLengthValue = childrenLength != null ? childrenLength.value : 0;
+            for (let i = 0; i < childrenLengthValue; i++) {
+              let child = this._getPropertyValue(childrenValue.properties, "" + i);
+              if (child instanceof Value) {
```

Comment:
I don't think you want to just run over it if it isn't a `Value`.
It can be a `null` if...
- It's an accessor property, or
- if it got deleted.
To be safe, I'd call `logError` with a message that indicates that this is an implementation limitation.